udata/frontend/helpers.py

```diff
@@ -361,3 +362,16 @@ def to_json(data):
 def format_number(number):
     '''A locale aware formatter.'''
     return format_number_babel(number, locale=g.lang_code)
+
+
+@front.app_template_filter()
+def group_resources_by_type(resources):
```

Comment:
As this is only used by the `datasets.show` page, this may be moved into `dataset/views.py` (you can use `blueprint.app_template_(filter|global)()` on any blueprint)

Reply:
BTW, aren't you rewriting the [`groupby` filter](http://jinja.pocoo.org/docs/2.10/templates/#groupby) ?

Reply:
Not really, I need to preserve order of `RESOURCE_TYPES` too. Ok for `dataset/views.py`.